### DISCUSSION TITLE
Any Lightbox options out there?

### DISCUSSION THREAD

**Discussion by andrewscofield** [igordanchenko/react-photo-album#68] [General]
Has anybody used this with a lightbox component?

**Comment by igordanchenko** [comment]
I used to provide a lightbox example in this repo, and you can still find it in the git history - https://githubbox.com/igordanchenko/react-photo-album/tree/v1.11.0/examples/lightbox

But I ended up deleting that demo since react-image-lightbox is no longer being maintained and doesn't work well with React 18 out of the box.

p.s. I'm actually considering putting together my own take on a modern lightbox component since all lightbox components I tried had some issues and were poorly maintained (or completely dormant)

**Comment by andrewscofield** [comment]
I would love a modern lightbox component for react. I could help out with it if you know the direct you want to head with it. My main questions would be, would it be standalone and work with react-photo-album? Would it use React Portals?

**Reply by igordanchenko** [reply]
It's going to be a standalone component, and it'll work seamlessly with react-photo-album. It'll use ReactDOM.createPortal by default. I'm about 80% done with what I'd like to see in v1, so I'm going to publish next week. Stay tuned!

**Reply by igordanchenko** [reply]
@andrewscofield, I've published [yet-another-react-lightbox](https://github.com/igordanchenko/yet-another-react-lightbox) v1 today. There is no documentation site yet, but I put together some [basic examples](https://codesandbox.io/s/yet-another-react-lightbox-examples-9qvmif?file=/src/examples/BasicExample.js) that should give a pretty good overview. Feedback and contributions are always welcome.

**Reply by andrewscofield** [reply]
Awesome, I'll try it out on a project I'm working on. As a NextJS user, I'm wondering already if there is a way to use a custom renderers, like on your `react-photo-album`. I'll check it out this weekend and get you some feedback if I have any.

**Reply by igordanchenko** [reply]
Yes, there are plenty of customization options. In addition to custom renderers, `yet-another-react-lightbox` also supports plugins, providing even greater customization opportunities.

**Comment by andrewscofield** [comment]
@igordanchenko I never got back to you but I did get this a go and it went really well! It was able to do everything I wanted so I pushed it to a small site I created for a client:  

https://www.stonemilldesign.com/services/bathrooms

There I am using https://github.com/igordanchenko/react-photo-album and https://github.com/igordanchenko/yet-another-react-lightbox together and its working great!

**Reply by igordanchenko** [reply]
@andrewscofield, thank you for your feedback!
